### PR TITLE
refactor(ci): Refactor of GHA CI files

### PR DESCRIPTION
**Pull request by S-Coyle**
Updated version of the checkout action being used to v2.
Tidied and condensed the caching.
Removed some unnecessary use of the cargo action, switching to
use the cargo command instead.

Supersedes #1232

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -122,46 +108,43 @@ jobs:
...
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-cache-${{ hashFiles('**/Cargo.lock') }}
+
+      # Test
+      - name: Immutable Data Tests
+        run: cargo test --manifest-path=safe_core/Cargo.toml --features=testing create_and_retrieve # id tests only
 
   # Test publish using --dry-run.
   # Only safe_core, as auth and app will fail if core was version-updated.
```
Comment (Yoga07) [inline .github/workflows/pr.yml:135]: We don't have auth and app any more here. So should be good to have this line removed :)

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -164,31 +164,30 @@ jobs:
...
-      - uses: actions-rs/cargo@v1
-        with:
-          command: publish
-          args: --manifest-path=safe_authenticator/Cargo.toml
+
+      - name: Publish safe_authenticator
         if: steps.versioning.outputs.auth_version != steps.versioning.outputs.old_auth_version
-     
\ No newline at end of file
+        run: cargo publish --manifest-path=safe_authenticator/Cargo.toml
```
Comment (Yoga07) [inline .github/workflows/master.yml:193]: Same, no more safe_authenticator in SCL. Better to have it removed too?

**Review by Yoga07 — COMMENTED** [review]
Just a couple of minor ones :)

File: .github/workflows/pr.yml (hunk 1 above)
Comment (S-Coyle) [reply .github/workflows/pr.yml]: Removed the `--manifest-path=safe_core/Cargo.toml` part of the test, thanks

File: .github/workflows/master.yml (hunk 2 above)
Comment (S-Coyle) [reply .github/workflows/master.yml]: removed, thanks

**Review by Yoga07 — APPROVED** [review]

**Pull request merged**